### PR TITLE
Rds templates, fixes to comply with best practices

### PR DESCRIPTION
**Pull request by marcvaneijk**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* replaced assetLocation variable with _artifactsLocation and _artifactsLocationSasToken parameters
* correct casing for _artifactsLocation defaultValue
* included trailing / for _artifactsLocation defaultValue
* added _artifactsLocationSasToken to concats
* moved DSC assets into DSC folder
* Updated timestamp in metadata.json
* lowercase for scripts folder

### CODE REVIEW

File: rds-deployment-existing-ad/azuredeploy.json [hunk 1]
```diff
@@ -448,7 +461,7 @@
                         "typeHandlerVersion": "2.11",
                         "autoUpgradeMinorVersion": true,
                         "settings": {
-                            "ModulesUrl": "[concat(variables('assetLocation'),'/Configuration.zip')]",
+                            "ModulesUrl": "[concat(parameters('_artifactsLocation'),'DSC/Configuration.zip',parameters('_artifactsLocationSasToken'))]",
```
Comment (bmoore-msft) [inline rds-deployment-existing-ad/azuredeploy.json]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#creating-uris for how to create the uris

File: rds-deployment-existing-ad/azuredeploy.json [hunk 2]
```diff
@@ -531,7 +544,7 @@
...
```
Comment (bmoore-msft) [inline rds-deployment-existing-ad/azuredeploy.json]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#creating-uris for how to create the uris

File: rds-deployment-existing-ad/azuredeploy.json [hunk 3]
```diff
@@ -613,7 +626,7 @@
                 "typeHandlerVersion": "2.11",
                 "autoUpgradeMinorVersion": true,
                 "settings": {
-                    "modulesUrl": "[concat(variables('assetLocation'),'/Configuration.zip')]",
+                    "modulesUrl": "[concat(parameters('_artifactsLocation'),'DSC/Configuration.zip',parameters('_artifactsLocationSasToken'))]",
```
Comment (bmoore-msft) [inline rds-deployment-existing-ad/azuredeploy.json]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#creating-uris for how to create the uris

File: rds-deployment-ha-gateway/azuredeploy.json [hunk 4]
```diff
@@ -347,7 +360,7 @@
                         "typeHandlerVersion": "2.76",
                         "autoUpgradeMinorVersion": true,
                         "settings": {
-                            "ModulesUrl": "[concat(variables('assetLocation'),'/Configuration.zip')]",
+                            "ModulesUrl": "[concat(parameters('_artifactsLocation'),'DSC/Configuration.zip',parameters('_artifactsLocationSasToken'))]",
```
Comment (bmoore-msft) [inline rds-deployment-ha-gateway/azuredeploy.json]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#creating-uris for how to create the uris

File: rds-deployment-ha-gateway/azuredeploy.json [hunk 5]
```diff
@@ -380,7 +393,7 @@
                         "autoUpgradeMinorVersion": true,
                         "settings": {
                             "fileUris": [
-                                "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/rds-deployment-ha-gateway/Scripts/WebAndGwFarmAdd_PostConfig1.1.ps1"
+                                "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/rds-deployment-ha-gateway/scripts/WebAndGwFarmAdd_PostConfig1.1.ps1"
```
Comment (bmoore-msft) [inline rds-deployment-ha-gateway/azuredeploy.json]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#creating-uris for how to create the uris

File: rds-deployment/azuredeploy.json [hunk 6]
```diff
@@ -82,23 +82,36 @@
...
     }
   },
   "variables": {
-    "adAssetLocation": "https://raw.githubusercontent.com/Azure/AzureStack-QuickStart-Templates/master/ad-non-ha",
     "adVMSize": "Standard_A1",
     "adVnetName": "[concat('ADVNET',resourceGroup().name)]",
     "adSubnetName": "[concat('ADStaticSubnet',resourceGroup().name)]",
     "staticSubnetID": "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('adVnetName'), variables('adSubnetName'))]",
-    "adTemplateURL": "[concat(variables('adAssetLocation'),'/adVmTemplate.json')]",
+    "adTemplateURL": "[concat(parameters('_artifactsLocation'),'adVmTemplate.json',parameters('_artifactsLocationSasToken'))]",
```
Comment (bmoore-msft) [inline rds-deployment/azuredeploy.json]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#creating-uris for how to create the uris

File: rds-deployment/azuredeploy.json [hunk 7]
```diff
@@ -82,23 +82,36 @@
...
     "adStorageName": "[tolower(concat('adsa',uniqueString(resourceGroup().id)))]",
     "adVmDeployment": "CreateAdVms",
     "adVmDeploymentId": "[concat('Microsoft.Resources/deployments/', variables('adVmDeployment'))]",
-    "deployPrimaryAdTemplateURL": "[concat(variables('adAssetLocation'),'/deployPrimaryAD.json')]",
+    "deployPrimaryAdTemplateURL": "[concat(parameters('_artifactsLocation'),'deployPrimaryAD.json',parameters('_artifactsLocationSasToken'))]",
```
Comment (bmoore-msft) [inline rds-deployment/azuredeploy.json]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#creating-uris for how to create the uris

File: rds-deployment/azuredeploy.json [hunk 8]
```diff
@@ -82,23 +82,36 @@
...
     "deployPrimaryAd": "DeployPrimaryAd",
     "deployPrimaryAdID": "[concat('Microsoft.Resources/deployments/', variables('deployPrimaryAd'))]",
     "adPDCVMName": "advm",
-    "vnetwithDNSTemplateURL": "[concat(variables('adAssetLocation'),'/vnet-with-dns-server.json')]",
+    "vnetwithDNSTemplateURL": "[concat(parameters('_artifactsLocation'),'vnet-with-dns-server.json',parameters('_artifactsLocationSasToken'))]",
```
Comment (bmoore-msft) [inline rds-deployment/azuredeploy.json]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#creating-uris for how to create the uris

File: rds-deployment/azuredeploy.json [hunk 9]
```diff
@@ -128,7 +141,7 @@
     "subnetAddressRange": "10.0.0.0/24",
     "dnsServerPrivateIp": "10.0.0.4",
     "subnet-id": "[concat(resourceId('Microsoft.Network/virtualNetworks', variables('adVnetName')),'/subnets/', variables('adSubnetName'))]",
-    "assetLocation": "https://raw.githubusercontent.com/Azure/azure-QuickStart-Templates/master/rds-deployment/",
+    "assetLocation": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/rds-deployment/",
```
Comment (bmoore-msft) [inline rds-deployment/azuredeploy.json]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#creating-uris for how to create the uris

File: rds-deployment/azuredeploy.json [hunk 10]
```diff
@@ -662,7 +675,7 @@
             "typeHandlerVersion": "2.11",
             "autoUpgradeMinorVersion": true,
             "settings": {
-              "modulesUrl": "[concat(variables('assetLocation'),'/Configuration.zip')]",
+              "modulesUrl": "[concat(parameters('_artifactsLocation'),'DSC/Configuration.zip',parameters('_artifactsLocationSasToken'))]",
```
Comment (bmoore-msft) [inline rds-deployment/azuredeploy.json]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#creating-uris for how to create the uris

File: rds-deployment/azuredeploy.json [hunk 11]
```diff
@@ -746,7 +759,7 @@
             "typeHandlerVersion": "2.11",
             "autoUpgradeMinorVersion": true,
             "settings": {
-              "ModulesUrl": "[concat(variables('assetLocation'),'/Configuration.zip')]",
+              "ModulesUrl": "[concat(parameters('_artifactsLocation'),'DSC/Configuration.zip',parameters('_artifactsLocationSasToken'))]",
```
Comment (bmoore-msft) [inline rds-deployment/azuredeploy.json]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#creating-uris for how to create the uris

File: rds-deployment/azuredeploy.json [hunk 12]
```diff
@@ -832,7 +845,7 @@
         },
         "publisher": "Microsoft.Powershell",
         "settings": {
-          "modulesUrl": "[concat(variables('assetLocation'),'/Configuration.zip')]",
+          "modulesUrl": "[concat(parameters('_artifactsLocation'),'DSC/Configuration.zip',parameters('_artifactsLocationSasToken'))]",
```
Comment (bmoore-msft) [inline rds-deployment/azuredeploy.json]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#creating-uris for how to create the uris

File: rds-update-certificate/azuredeploy.json [hunk 13]
```diff
@@ -114,7 +114,7 @@
           "fileUris": [
             "[variables('impersonateScript')]",
             "[variables('setPublishedNameScript')]",
-            "[concat(parameters('_artifactsLocation'), '/', variables('scriptFolder'), '/', variables('scriptFileName'), parameters('_artifactsLocationSasToken'))]"
+            "[concat(parameters('_artifactsLocation'), variables('scriptFolder'), '/', variables('scriptFileName'), parameters('_artifactsLocationSasToken'))]"
```
Comment (bmoore-msft) [inline rds-update-certificate/azuredeploy.json]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#creating-uris for how to create the uris

File: rds-update-rdsh-collection/azuredeploy.json [hunk 14]
```diff
@@ -227,7 +227,7 @@
               "typeHandlerVersion": "2.11",
               "autoUpgradeMinorVersion": true,
               "settings": {
-                "ModulesUrl": "[concat(parameters('_artifactsLocation'),'/DSC/Configuration.zip', parameters('_artifactsLocationSasToken'))]",
+                "ModulesUrl": "[concat(parameters('_artifactsLocation'),'DSC/Configuration.zip', parameters('_artifactsLocationSasToken'))]",
```
Comment (bmoore-msft) [inline rds-update-rdsh-collection/azuredeploy.json]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#creating-uris for how to create the uris

File: rds-update-rdsh-collection/azuredeploy.json [hunk 15]
```diff
@@ -265,7 +265,7 @@
           "settings": {
             "fileUris": [
               "[variables('impersonateScript')]",
-              "[concat(parameters('_artifactsLocation'), '/', variables('scriptFolder'), '/', variables('scriptFileName'), parameters('_artifactsLocationSasToken'))]"
+              "[concat(parameters('_artifactsLocation'), variables('scriptFolder'), '/', variables('scriptFileName'), parameters('_artifactsLocationSasToken'))]"
```
Comment (bmoore-msft) [inline rds-update-rdsh-collection/azuredeploy.json]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#creating-uris for how to create the uris

File: rds-deployment-existing-ad/azuredeploy.json (hunk 1 above)
Comment (marcvaneijk) [reply rds-deployment-existing-ad/azuredeploy.json]: fixed

File: rds-deployment-existing-ad/azuredeploy.json (hunk 2 above)
Comment (marcvaneijk) [reply rds-deployment-existing-ad/azuredeploy.json]: fixed

File: rds-deployment-existing-ad/azuredeploy.json (hunk 3 above)
Comment (marcvaneijk) [reply rds-deployment-existing-ad/azuredeploy.json]: fixed

File: rds-deployment-ha-gateway/azuredeploy.json (hunk 4 above)
Comment (marcvaneijk) [reply rds-deployment-ha-gateway/azuredeploy.json]: fixed

File: rds-deployment-ha-gateway/azuredeploy.json (hunk 5 above)
Comment (marcvaneijk) [reply rds-deployment-ha-gateway/azuredeploy.json]: fixed

File: rds-deployment/azuredeploy.json (hunk 6 above)
Comment (marcvaneijk) [reply rds-deployment/azuredeploy.json]: fixed

File: rds-deployment/azuredeploy.json (hunk 7 above)
Comment (marcvaneijk) [reply rds-deployment/azuredeploy.json]: fixed

File: rds-deployment/azuredeploy.json (hunk 8 above)
Comment (marcvaneijk) [reply rds-deployment/azuredeploy.json]: fixed

File: rds-deployment/azuredeploy.json (hunk 9 above)
Comment (marcvaneijk) [reply rds-deployment/azuredeploy.json]: fixed

File: rds-deployment/azuredeploy.json (hunk 10 above)
Comment (marcvaneijk) [reply rds-deployment/azuredeploy.json]: fixed

File: rds-deployment/azuredeploy.json (hunk 11 above)
Comment (marcvaneijk) [reply rds-deployment/azuredeploy.json]: fixed

File: rds-deployment/azuredeploy.json (hunk 12 above)
Comment (marcvaneijk) [reply rds-deployment/azuredeploy.json]: fixed

File: rds-update-certificate/azuredeploy.json (hunk 13 above)
Comment (marcvaneijk) [reply rds-update-certificate/azuredeploy.json]: fixed

File: rds-update-rdsh-collection/azuredeploy.json (hunk 14 above)
Comment (marcvaneijk) [reply rds-update-rdsh-collection/azuredeploy.json]: fixed

File: rds-update-rdsh-collection/azuredeploy.json (hunk 15 above)
Comment (marcvaneijk) [reply rds-update-rdsh-collection/azuredeploy.json]: fixed

**Comment by marcvaneijk** [comment]
Thanks @bmoore-msft 
I've updated all refs to uri.
Concerning the link to the Azure Stack repo. I couldn't find the same template in this repo. Would you like me to copy it here or keep the defaultValue pointing to the Azure Stack repo?

**Comment by bmoore-msft** [comment]
Check the CI errors, there are a few metadata.json failures as well...

re: the AzureStack cross link sample - looks like it just needs AD?  Is that just this: https://github.com/Azure/azure-quickstart-templates/tree/master/active-directory-new-domain

**Comment by marcvaneijk** [comment]
Fixed metadata CI errors.
Mapping it to the AD template it this repo requires significant changes to the RDS template (existing vnet, merging linked templates and paramater mapping). I'll give it a shot.

**Comment by bmoore-msft** [comment]
branch is gone

**Pull request closed without merging**